src/com/maddyhome/idea/vim/helper/SearchHelper.java

```diff
@@ -477,7 +477,7 @@ private static boolean checkInString(@NotNull CharSequence chars, int pos, boole
 
     boolean inString = false;
     boolean inChar = false;
-    for (int i = offset; i <= pos; i++) {
+    for (int i = offset; i <= pos && chars.length()>i; i++) {
```

Comment:
It would be enough to check that `pos` is in range of `chars` once before the loop.